delfin/drivers/huawei/oceanstor/oceanstor.py

```diff
@@ -354,6 +354,146 @@ def list_disks(self, context):
             LOG.error(err_msg)
             raise exception.InvalidResults(err_msg)
 
+    def list_filesystems(self, context):
+        try:
+            # Get list of OceanStor filesystems details
+            fss = self.client.get_all_filesystems()
+
+            fs_list = []
+            worm_type = {
+                consts.FS_WORM_COMPLIANCE: constants.WORMType.COMPLIANCE,
+                consts.FS_WORM_AUDIT_LOG: constants.WORMType.AUDIT_LOG,
+                consts.FS_WORM_ENTERPRISE: constants.WORMType.ENTERPRISE
+            }
+            for fs in fss:
+                status = constants.FilesystemStatus.FAULTY
+                if fs['HEALTHSTATUS'] == consts.FS_HEALTH_NORMAL:
+                    status = constants.FilesystemStatus.NORMAL
+                type = constants.FSType.THICK
+                if fs['ALLOCTYPE'] == consts.FS_TYPE_THIN:
+                    type = constants.FSType.THIN
+
+                sector_size = int(fs['SECTORSIZE'])
+                total_cap = int(fs['CAPACITY']) * sector_size
+                used_cap = int(fs['ALLOCCAPACITY']) * sector_size
+
+                compressed = False
+                if fs['ENABLECOMPRESSION'] != 'false':
+                    compressed = True
+
+                deduplicated = False
+                if fs['ENABLEDEDUP'] != 'false':
+                    deduplicated = True
+
+                fs = {
```

Comment:
please update hether security_mode is applicable for fs in oceanstore or not?

Reply:
This will be handled in separate issue